### PR TITLE
Adding chrono DateTime<Local> support for postgres

### PR DESCRIPTION
**Pull request by noyez**

### CODE REVIEW

**Comment by sgrif** [comment]
Please run rustfmt

File: diesel/src/pg/types/date_and_time/chrono.rs [hunk 1]
```diff
@@ -64,6 +64,13 @@ impl FromSql<Timestamptz, Pg> for DateTime<Utc> {
     }
 }
 
+impl FromSql<Timestamptz, Pg> for DateTime<Local> {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        let naive_date_time = <NaiveDateTime as FromSql<Timestamptz, Pg>>::from_sql(bytes)?;
+        Ok(DateTime::from_utc(naive_date_time, *Local::now().offset()))
```
Comment (sgrif) [inline diesel/src/pg/types/date_and_time/chrono.rs]: What do you think about this instead? `Local::from_utc_datetime(&naive_date_time)`

Comment (noyez) [reply diesel/src/pg/types/date_and_time/chrono.rs]: sure, why not? Change to use `Local::from_utc_datetime` pushed.

File: diesel/src/lib.rs [hunk 2]
```diff
@@ -110,7 +110,7 @@
                        note = "The x128-column-tables feature has been reanmed to 128-column-tables. The x was a workaround for a bug in crates.io that has since been resolved"))]
 #![cfg_attr(feature = "unstable", feature(specialization, try_from))]
 // Built-in Lints
-#![deny(warnings, missing_debug_implementations, missing_copy_implementations, missing_docs)]
+#![warn(missing_debug_implementations, missing_copy_implementations, missing_docs)]
```
Comment (sgrif) [inline diesel/src/lib.rs]: This line should not be changed.

**Comment by noyez** [comment]
Trying to fix lineage of PR. It was based on `master` however it should be tracking the current release branch, which is currently `1.3.x`. Rookie mistake on my part.

**Comment by weiznich** [comment]
This should continue to target master. Only backports of bugfixes should target release branches. New features belonging into the master branch.

**Comment by noyez** [comment]
Ok. Based off master again. Sorry for the extraneous changes.

**Pull request closed without merging**